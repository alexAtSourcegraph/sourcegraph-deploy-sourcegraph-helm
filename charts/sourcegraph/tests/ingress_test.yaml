suite: ingress
templates:
  - frontend/sourcegraph-frontend.Ingress.yaml
release:
  name: sourcegraph
  namespace: sourcegraph
tests:
  - it: should render ingress with only one host
    set:
      frontend:
        ingress:
          enabled: true
          host: "chart-example.local"
    asserts:
      - matchSnapshot:
          path: spec.rules
  - it: should render ingress with many hosts when extraHosts is set
    set:
      frontend:
        ingress:
          enabled: true
          host: "chart-example.local"
          extraHosts:
            - chart-example-1.local
            - chart-example-2.local
    asserts:
      - matchSnapshot:
          path: spec.rules
  - it: should render ingress with a one tls secret
    set:
      frontend:
        ingress:
          enabled: true
          host: "chart-example.local"
          tlsSecret: "chart-example-tls"
    asserts:
      - matchSnapshot:
          path: spec.tls
  - it: should render ingress with many tls secrets when extraTLSSecrets is set
    set:
      frontend:
        ingress:
          enabled: true
          host: "chart-example.local"
          tlsSecret: "chart-example-tls"
          extraTLSSecrets:
            - secretName: chart-example-tls-1
              hosts:
                - chart-example-1.local
            - secretName: chart-example-tls-2
              hosts:
                - chart-example-2.local
    asserts:
      - matchSnapshot:
          path: spec.tls
