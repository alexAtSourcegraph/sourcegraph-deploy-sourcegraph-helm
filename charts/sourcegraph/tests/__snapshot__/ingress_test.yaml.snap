should render ingress with a one tls secret:
  1: |
    - hosts:
      - chart-example.local
      secretName: chart-example-tls
should render ingress with many hosts when extraHosts is set:
  1: |
    - host: chart-example.local
      http:
        paths:
        - backend:
            service:
              name: sourcegraph-frontend
              port:
                number: 30080
          path: /
          pathType: Prefix
    - host: chart-example-1.local
      http:
        paths:
        - backend:
            service:
              name: sourcegraph-frontend
              port:
                number: 30080
          path: /
          pathType: Prefix
    - host: chart-example-2.local
      http:
        paths:
        - backend:
            service:
              name: sourcegraph-frontend
              port:
                number: 30080
          path: /
          pathType: Prefix
should render ingress with many tls secrets when extraTLSSecrets is set:
  1: |
    - hosts:
      - chart-example.local
      secretName: chart-example-tls
    - hosts:
      - chart-example-1.local
      secretName: chart-example-tls-1
    - hosts:
      - chart-example-2.local
      secretName: chart-example-tls-2
should render ingress with only one host:
  1: |
    - host: chart-example.local
      http:
        paths:
        - backend:
            service:
              name: sourcegraph-frontend
              port:
                number: 30080
          path: /
          pathType: Prefix
